WordPress/src/main/java/org/wordpress/android/ui/menus/items/MenuItemEditorFactory.java

```diff
@@ -42,22 +80,55 @@ else if (typeName.compareToIgnoreCase("post_type") == 0) {
             else if (typeName.compareToIgnoreCase("jetpack-testimonial") == 0) {
                 return JETPACK_TESTIMONIAL;
             }
+            else if (typeName.compareToIgnoreCase("jetpack-portfolio") == 0) {
+                return JETPACK_PORTFOLIO;
+            }
+            else if (typeName.compareToIgnoreCase("jetpack-comic") == 0) {
+                return JETPACK_COMIC;
+            }
 
             else return NULL;
         }
 
-        public static ITEM_TYPE typeForIndex(int index){
-            if (NULL.ordinal() == index) return NULL;
-            if (POST.ordinal() == index) return POST;
-            if (PAGE.ordinal() == index) return PAGE;
-            if (CATEGORY.ordinal() == index) return CATEGORY;
-            if (TAG.ordinal() == index) return TAG;
-            if (LINK.ordinal() == index) return LINK;
-            if (JETPACK_TESTIMONIAL.ordinal() == index) return JETPACK_TESTIMONIAL;
-            if (CUSTOM.ordinal() == index) return CUSTOM;
+        public static String nameForItemType(Context context, ITEM_TYPE itemType) {
+            String name = "";
 
-            return NULL;
+            switch (itemType) {
+                case POST:
+                    name = context.getString(R.string.menu_item_type_post);
+                    break;
+                case PAGE:
+                    name = context.getString(R.string.menu_item_type_page);
+                    break;
+                case CATEGORY:
+                    name = context.getString(R.string.menu_item_type_category);
+                    break;
+                case TAG:
+                    name = context.getString(R.string.menu_item_type_tag);
+                    break;
+                case LINK:
+                    name = context.getString(R.string.menu_item_type_link);
+                    break;
+                case JETPACK_TESTIMONIAL:
+                    name = context.getString(R.string.menu_item_type_testimonial);
+                    break;
+                case JETPACK_PORTFOLIO:
+                    name = context.getString(R.string.menu_item_type_portfolio);
+                    break;
+                case JETPACK_COMIC:
+                    name = context.getString(R.string.menu_item_type_comic);
+                    break;
+                case NULL:
```

Comment:
These two cases (`NULL` and `CUSTOM`) are essentially no-ops since name is initialized as an empty string. We could remove unless there's a future use for them.